### DISCUSSION TITLE
Can Vercel host a Java Spring web app?

### DISCUSSION THREAD

**Discussion by JSONRice** [vercel/community#98] [Help] [answered]
I have a Java Spring web app for a D&D pilot:

https://github.com/jasonwr/rpg-dnd-server

Just wanting to know if deploying a Java Spring Boot app is even possible in Vercel. I was able to deploy it fine in Heroku but have a client interfacing app running on Vercel and would have to get the server interfacing app running in Vercel as well. Thank you in advance.

**Comment by leerob** [comment, marked as answer]
Hey! No, currently it is not possible to deploy a Java Spring Boot application. Sorry about that.

**Reply by Dels1x** [reply]
Do you plan on providing support for it in future?

**Comment by aysenurr41** [comment]
Hİ , I have a java spring web app. 
How would I do it on Heroku just to deploy a Java Spring Boot application.
can you help me ? :)

**Reply by japzio** [reply]
you can use maven plugin https://devcenter.heroku.com/articles/deploying-java-applications-with-the-heroku-maven-plugin or heroku cli https://devcenter.heroku.com/categories/command-line

**Reply by BLemine** [reply]
You can do that either with a docker container or source code, check their doc from here https://devcenter.heroku.com/articles/deploying-spring-boot-apps-to-heroku

**Comment by ederbiason** [comment]
Vercel still don't have support for Java Spring Boot apps? Heroku will stop to work for free in november, do you guys knows an alternative? For free...

**Reply by amyegan** [reply]
Spring Boot documentation has a list of cloud deployment options: https://docs.spring.io/spring-boot/docs/current/reference/html/deployment.html#deployment.cloud

**Reply by ederbiason** [reply]
Thanks!!

**Reply by adbci** [reply]
I was facing same issue than you.
Went for Render (render.com)
It's just perfect as a replacement.
Just a bit slow with free plan, but paid plan are much cheaper than Heroku's.

You should give it a try :)

(FYI, I'm not part of Render board nor team 😅)

**Reply by ederbiason** [reply]
Oh, I will take a look! Heroku will stop the free plan this month, so I will try Render. Thanks for the tip

**Reply by japzio** [reply]
railway.app https://docs.railway.app/deploy/builds

**Comment by MilanObrenovic** [comment]
So what am i supposed to do if i build backend in spring boot but deploy my frontend with vercel?

**Reply by japzio** [reply]
no other choice it think, run it with other paas such as mentioned here https://github.com/orgs/vercel/discussions/98#discussioncomment-3610945, and r[ailway.app](https://docs.railway.app/deploy/builds) (https://nixpacks.com/docs/providers/java)

**Reply by Kordedekehine** [reply]
I'm just trying it for the first time and it keeps failing
